### ISSUE TITLE
Graph Debugger 

### ISSUE THREAD

**Issue by josvanroosmalen** [marklogic-community/pipes#139]
It would be nice to have a "graph debugger". 

Implementation idea.

- Run preview
- When you close the preview window you can hoover over the nodes, and see the input/output values of the node of the last run. 
- In case of a crash (in the previewer) it should still be possible to view the input/outputs of the evaluated nodes prior to the crash. 

I am not sure it is possible to capture the `setOutputData` values so that it later can be displayed in tooltips.

**Comment by sashamitrovich** [comment]
Hi @josvanroosmalen, other than it would be nice (which I agree), any other reasons for the debugger? 

Which problem/difficulty/challenge is the ticket addressing?

Could you rephrase it to the following format: "As a Pipes user, I'd like to be able to see input/out values of each block in the graph (for a given custom step input document) so that I can better understand why the graph is producing a certain result"? Or something similar...
